### PR TITLE
MonoComonad instance for MinLen (Succ (Succ a))

### PR DESCRIPTION
**Pull request by mstksg**
In truth this is a fully compatible generalization for `MinLen (Succ Zero)`, but the `MinLen (Succ Zero)` instance is much more efficient and simple I think.  I toyed around with both having the same instance and then using runtime checks on reified types, but it seemed a bit odd to not take advantage of type erasure at compile type for static dispatch.

If you think they should be merged, I can merge them back too.

It might also be that the generalized instance is not worth putting in at all, because it has unexpected consequences: As a consequence`extend osum` on a `MinLen 3 [Int]` and `extend osum` on a `MinLen 5 [Int]` have very different results, because the type of `osum` is different; one is `osum :: MinLen 3 [Int] -> Int`, and the other is `osum :: MinLen 5 [Int] -> Int`.  So their extended versions are different, but it's not obvious at first.  Maybe not having the extra instances would prevent this unexpected behavior at the sacrifice of not generalizing the instance.

### CODE REVIEW

File: src/Data/MinLen.hs [hunk 1]
```diff
@@ -482,17 +483,74 @@ minimumBy :: MonoFoldable mono
...
                                 . flip evalState mono
                                 . ofor mono
                                 . const
                                 . state
                                 $ \mono' -> (f (MinLen mono'), tailEx mono')
+
+-- | 'oextract' is 'head'.
+--
+-- Similar to the instance for 'NonNull', but when the lower length limit
+-- is reached, instead of 'tail'ing, succesively "rotates left" the 'mono'.
```
Comment (kbillings) [inline src/Data/MinLen.hs]: Misspelled `succesively` -> `successively`

**Comment by snoyberg** [comment]
I don't feel strongly about it, but I think I'm -1 on the extra instance. It leads to a strange situation where `oextract` behaves quite differently depending on the *annotated* minimum length of the container, which doesn't feel right. That said, I haven't ever actually used Comonads myself, so my thoughts should be taken with a large grain of salt.

**Comment by mstksg** [comment]
I too am against it :)

**Comment by snoyberg** [comment]
The rest of the doc improvements though in this PR look pretty good. Interested in removing the instance and then I can merge the rest of the changes?

**Comment by mstksg** [comment]
Instance removed as of now, d6a522b :)

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**